driver/bpf/fillers.h

```diff
@@ -571,236 +571,158 @@ FILLER(sys_poll_x, true)
 	return bpf_poll_parse_fds(data, false);
 }
 
-#define MAX_IOVCNT 32
-#define MAX_IOVCNT_COMPAT 8
+#ifdef CONFIG_COMPAT
```

Comment:
Only real downside of the patch (except for its ugliness probably :D ) is that we will have max iovec count of 8 when config compat is enabled; this is needed because we must have a constant to be able to unroll the loops below. Not a big deal; this is only on the old ebpf probe.